src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -81,6 +84,23 @@ object UpdateEventItem {
     )
 }
 
+sealed trait RetrieveEvents
+case class EventId(id: Long) extends RetrieveEvents
+case class EventExternalId(externalId: String) extends RetrieveEvents
+
+object RetrieveEventCoders {
+  implicit val encodeRetrieveEvents: Encoder[RetrieveEvents] = new Encoder[RetrieveEvents] {
+    override def apply(a: RetrieveEvents): Json =
+      Json.obj(a match {
```

Comment:
I'm not sure that it's valid to mix the two types of retrieval in one request, but maybe...? Have we tested that?

Reply:
Looks like it's allowed in 0.6, but not in v1: https://doc.cognitedata.com/api/v1/#operation/getEventsByIds
I think we can just not bother with predicate pushdown for externalId for now.

Reply:
I saw that the python-sdk allows for retrieval by id and external id. https://cognite-sdk-python.readthedocs-hosted.com/en/latest/cognite.html#retrieve-multiple-events-by-id